dj_search/search.py

```diff
@@ -0,0 +1,143 @@
+import datajoint as dj
+import re
+import inspect
+from termcolor import colored
+
+
+class DJSearch:
+
+    def __init__(self, db_prefix='', context=None):
+        self.context = context or inspect.currentframe().f_back.f_locals
+        self.virtual_modules = {}
+
+        self.schema_names = [s for s in dj.list_schemas() if s.startswith(db_prefix)]
+
+        tbls_defi = []
+        for schema_name in self.schema_names:
+            self.virtual_modules[schema_name] = dj.create_virtual_module(schema_name, schema_name)
+            schema_defi = self.virtual_modules[schema_name].schema.save()
+            schema_defi = schema_defi.replace('@schema', '@{}'.format(schema_name))
+
+            for match in re.finditer("VirtualModule\('(\w+)', '(\w+)'\)", schema_defi):
+                vmod, vmod_rep = match.groups()
+                schema_defi = schema_defi.replace(vmod, vmod_rep)
+
+            tbls_defi.append(schema_defi)
+
+        # join definitions from all schema, remove INDEX and UNIQUE lines
+        defi = '\n'.join(tbls_defi)
+        defi = re.sub('\s+?INDEX.+?\n|\s+?UNIQUE.+?\n', '', defi, flags=re.MULTILINE)
+        defi = re.sub('([\(\)\[\]\w])( *)"""', '\g<1>\n\g<2>"""', defi)
+
+        self.definition_string = defi
+
+    def search(self, search_str, level=None):
+        """
+        :param search_str: string to search for
+        :param level: 'table', 'attribute', 'comment'
+        :return:
+        """
+        m = DJMatch(search_str, self.definition_string, self.virtual_modules, level=level)
```

Comment:
Why do we need two classes to do a similar job? What's the problem combining them to one?

Reply:
`DJMatch` is an object with the matched search-string, returning the tables (and their definition) with the matched string. Users can access these tables as well, through the returned `DJMatch` object. And we can potentially add more feature here.

One `DJSearch` object can potentially return different DJMatch objects that users can retain and access more flexibly